### ISSUE TITLE
Replace text UI doesn't have correct focus on page load

### ISSUE THREAD

**Issue by leppert** [harvard-lil/h2o#439]
Recreate:

- Visit a case
- Highlight text
- Click "Replace..."

When the page reloads, a box will say "Enter text to replace..." and appear to have focus, but key presses will have no effect until the spacebar is pressed, at which point proper focus is established and text can be entered. Tested in Chrome 67 and Safari 11.

![image](https://user-images.githubusercontent.com/26643/42057947-dbad9a9e-7aec-11e8-85c8-a52e4d5b8185.png)

**Comment by rebeccacremona** [comment]
Ah yes, this is confusing. I'm almost certain, though, that focus itself is actually correct: I recently added code that would programmatically set the focus to a desired annotation on page load/re-load. Beforehand, the viewport would remain scrolled to keep the annotation in visual focus (on particular browsers), but programmatic focus would still be up at the top of the page in all browsers, and the viewport would be visibly scrolled to the top in other browsers. In that world, where programmatic focus was not set, the annotation was visibly singled out on page refresh, so it was natural for the user, after finding the new elision, to click the words "enter text to replace" to activate the contenteditable span and then edit the text. As it stands now, you still have to click/press to edit, just like before my changes, but now, the fact programmatic focus lands where it belongs makes that non-intuitive, instead of natural.

**Comment by leppert** [comment]
@rebeccacremona Any reason we shouldn't cut out that click by putting focus on the `contenteditable` span rather than the `.annotate.replacement` wrapper span?

```
<span role="button" tabindex="0" aria-expanded="false" class="annotate replacement" data-annotation-id="354367"><span class="text" data-annotation-id="354367" contenteditable="true">Can focus go here?</span></span>
```

**Comment by rebeccacremona** [comment]
@leppert Hmmm. Maybe? So that click/press triggers a bit of logic: 
```
function handleReplaceButtonPressed(e){
  if (isEditable()) {
    editAnnotationHandle(e.target.previousElementSibling);
    stagePreviousContent(e.target.innerText);
    setFocus(e.target.firstElementChild);
}
```
https://github.com/harvard-lil/h2o/blob/master/app/assets/javascripts/lib/ui/content/annotations/replace.js#L32. 

If one were to set the contenteditable span to have a tabindex of ="-1" so it can programmatically receive focus (but not be tabbed to), and then had an event listener that ran those same functions when the said span receives focus, that could work. One might need to tweak the logic that makes sure pressing space/enter while editing doesn't accidentally press the button... though maybe that will still work fine....! \shrugs. 

It might be less fiddly to add a special case to the code that is currently setting the focus (https://github.com/harvard-lil/h2o/blob/master/app/assets/javascripts/lib/ui/content/annotations/index.js#L18). Despite the comment, which is wrong lol, that also handles the other kinds of annotations (plain elisions, notes, highlights, and links), and it handles focus for both authors and non-logged-in users. I'm not that familiar with the author-facing side of things, but I -think- this is the only special case, where the button should also be pressed on page load (or the inner span focused as per your idea). Maybe easier to add something like "if isEditable() and if I'm a replacement, then click me" there?

I don't THINK I'm over-complicated things 😄. Maybe I am? I don't think so.

**Comment by rebeccacremona** [comment]
Actually, even just setting the focus directly to the inner span, in this one special case, would require a special case in https://github.com/harvard-lil/h2o/blob/master/app/assets/javascripts/lib/ui/content/annotations/index.js#L18, right?

**Comment by leppert** [comment]
@rebeccacremona `handleReplaceButtonPressed` doesn't appear to be called when tabbing into the contenteditable field, though the functionality (at least typing in text and saving it; not sure what else should be / is going on there) works as intended. What am I missing or misunderstanding in that case? Or is that simply a bug?

**Comment by rebeccacremona** [comment]
@leppert Ooh super interesting! Maybe those inner functions are superfluous! I've never looked at them before: a glance at `editAnnotationHandle` and `stagePreviousContent` suggests to me that they are meant to handle the situation where you are editing several replacement annotations consecutively, but just quickly messing around, I can't seem to find any problems just tabbing straight into multiple contenteditable spans and editing them either. Innnnnteresting.

**Comment by rebeccacremona** [comment]
Okay, so clearly I have nothing further to contribute here LOL.

**Comment by cgruppioni** [comment]
Has been fixed
